events/simple.go

```diff
@@ -11,26 +11,39 @@
 package events
 
 type simpleChannel[T any] struct {
-	subscribers         []chan T
-	subscriptionChannel chan chan T
-	unsubscribeChannel  chan chan T
-	eventChannel        chan T
-	eventBufferSize     int
-	closeChannel        chan struct{}
-	isClosed            bool
+	subscribers []chan T
+	// commandChannel manages all commands sent to this simpleChannel.
+	//
+	// It is important that all stuff gets sent through this single channel to ensure
+	// that the order of operations is preserved.
+	//
+	// WARNING: This does mean that non-event commands can block the database if the buffer
+	// size is breached (e.g. if many subscribe commands occupy the buffer).
+	commandChannel  chan any
+	eventBufferSize int
+	hasClosedChan   chan struct{}
+	isClosed        bool
 }
 
-// NewSimpleChannel creates a new simpleChannel with the given subscriberBufferSize and
+type subscribeCommand[T any] Subscription[T]
```

Comment:
Note: Fred suggested dropping the `command` suffix for these, or shortening it to `cmd`.  I have a minor preference for the current names (I like it being clear that they are commands, and not an e.g. `subscribe` type, I also prefer the more verbose name).

Happy to change this though if others want, or if Fred's preference is/becomes stronger than I think it is :)

Reply:
I would be against dropping the suffix because at the moment it clearly shows that these types belong to a single concept. Shortening would be fine though especially because `cmd` is well-known. But my personal preference is to keep `command`

Reply:
Prefer to keep it the way it is.

Reply:
Will keep as-is then :)